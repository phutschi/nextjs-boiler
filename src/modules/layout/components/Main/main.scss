*, html {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}

a {
  color: #000;
  cursor: pointer;
  margin: 0 2px;
  text-decoration: none;

  &:hover {
    background-image: linear-gradient(120deg, #cfd9df 0%, #e2ebf0 100%);
    background-repeat: no-repeat;
    background-size: 100% .1em;
    background-position: 0 103%;
  }

  &:active {
    color: #555;
    background-image: linear-gradient(120deg, #11998e 0%, #11998e 100%);
    background-repeat: no-repeat;
    background-size: 100% .1em;
    transition: all 0.15s ease-in;
  }
}

.main {
  margin: 0;

  &.primary {
    background-color: white;
    color: black;
  }

  &.inverted {
    background-color: black;
    color: white;
  }
}
