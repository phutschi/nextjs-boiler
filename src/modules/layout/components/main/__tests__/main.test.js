/* global expect */

import * as React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Main from '../';

describe( '<Main />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <Main /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Renders without props', () => {
        const wrapper = shallow( <Main /> );

        expect( wrapper.find( 'div.main' ) ).toHaveLength( 1 );
        expect( wrapper.find( 'div.main' ).hasClass( 'primary' ) ).toEqual( false );
        expect( wrapper.find( 'div.main' ).hasClass( 'inverted' ) ).toEqual( false );
    } );

    it( 'Renders with props', () => {
        const wrapper = shallow( <Main primary /> );

        expect( wrapper.find( 'div.main' ) ).toHaveLength( 1 );
        expect( wrapper.find( 'div.main' ).hasClass( 'primary' ) ).toEqual( true );
    } );

    it( 'Renders with child components', () => {
        const wrapper = shallow(
            <Main>
                <div className="main-child" />
            </Main>,
        );

        expect( wrapper.find( 'div.main' ) ).toHaveLength( 1 );
        expect( wrapper.find( 'div.main-child' ) ).toHaveLength( 1 );
    } );
} );
