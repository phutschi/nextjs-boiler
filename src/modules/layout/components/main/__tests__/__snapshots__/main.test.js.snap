// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Snapshot 1`] = `
<div
  className="jsx-1439403121 main "
/>
`;
